unumpy/tests/test_numpy.py

```diff
@@ -318,6 +319,6 @@ def test_ufuncs_results(backend, method, args, kwargs, res):
         pytest.xfail(reason="The backend has no implementation for this ufunc.")
 
     if isinstance(ret, sparse.SparseArray):
-        assert onp.equal(ret.todense(), res.todense())
+        ret, res = ret.todense(), res.todense()
 
-    assert onp.equal(ret, res)
+    assert np.all(np.equal(ret, res) or (np.isnan(ret) and np.isnan(res)))
```

Comment:
There's a reason I used `|` and `&`: They work on arrays element-wise. `or` and `and` attempt to make the input a Python `bool` before proceeding.

Reply:
Which won't succeed because only 0D arrays collapse to a `bool`.